TwigJs/Compiler/RoutingFunctionCompiler.php

```
@@ -0,0 +1,226 @@
+     * Router
+     * 
+     * @var RouterInterface
+     */
+    protected $router;
+
+    /**
+     * @param RouterInterface $router Router
+     */
+    public function __construct(RouterInterface $router)
+    {
+        $this->router = $router;
+    }
+
+    /**
+     * Compiles a twig function for use in javascript and compiles a method call
+     * appropriate for the specified node.
+     * 
+     * @param JsCompiler                    $compiler Javascript twig compiler
+     * @param Twig_Node_Expression_Function $node     Node for compilation
+     */
+    public function compile(JsCompiler $compiler, Twig_Node_Expression_Function $functionNode)
+    {
+        // Function arguments
+        $argumentNodes = iterator_to_array($functionNode->getNode('arguments'));
+
+        // Ensure that the argument list is not empty
+        if (empty($argumentNodes)) {
+            throw new \RuntimeException(sprintf(
+                "Can't compile %s function for javascript when no arguments are provided.",
+                $functionNode->getAttribute('name')
+            ));
+        }
+
+        // First argument will be the route
+        $routeNode = array_shift($argumentNodes);
+
+        // Only text nodes are supported as the routing path
+        if ($routeNode instanceof Twig_Node_Expression_Constant) {
+            $routeName = $routeNode->getAttribute('value');
+        } else {
+            $compiler
+                ->raw("(function(){throw ")
+                ->string(sprintf(
+                    "Only static route names can be compiled for javascript %s. Node type '%s' is not supported.",
+                    $node->getAttribute('name'),
+                    get_class($routeNode)
+                ))
+                ->raw(";})()")
+            ;
+            return;
+        }
+
+        // Get the route name from the node
+        $route = $this->router->getRouteCollection()->get($routeName);
+
+        // When no route exists throw an exception
```

I was previously throwing a `RuntimeException` however it was being caught and silently ignored. I'll add it back in, but I can't test that it's behaving correctly.